### ISSUE TITLE
gpg: key 3804BB82D39DC0E3: new key but contains no user ID - skipped

### ISSUE THREAD

**Issue by x-yuri** [rvm/rvm#5121]
## Description

Something is wrong with `409B6B1796C275462A1703113804BB82D39DC0E3` on `keys.openpgp.org`, which is [the recommended][a] and [the default keyserver][b] these days.

## Steps to reproduce

Debian 10:

```
$ gpg --keyserver keys.openpgp.org --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
gpg: key 3804BB82D39DC0E3: new key but contains no user ID - skipped
gpg: Total number processed: 1
gpg:           w/o user IDs: 1
```

Arch Linux:

```
$ gpg --keyserver keys.openpgp.org --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
gpg: key 3804BB82D39DC0E3: no user ID
gpg: Total number processed: 1
```

## Expected behavior

It gets imported.

## Actual behavior

It fails.

[a]: https://unix.stackexchange.com/questions/656205/sks-keyservers-gone-what-to-use-instead/656210#656210
[b]: https://lists.gnupg.org/pipermail/gnupg-users/2021-June/065261.html

**Comment by Al3nMicL** [comment]
Try the mit keyserver instead. See page [rvm security](https://rvm.io/rvm/security) for details.

**Comment by x-yuri** [comment]
Indeed, once I was able to retrieve the keys from hkp://pgp.mit.edu, but then:

```
$ gpg --keyserver hkp://pgp.mit.edu --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
gpg: keyserver receive failed: No keyserver available
```

More importantly, isn't something wrong with the key published at keys.openpgp.org?

**Comment by x-yuri** [comment]
From what I can see keys.openpgp.org has stricter standards. It doesn't make the owner's email address publicly available unless the owner consented to this. And as such such a key can't be imported from this key server.

https://keys.openpgp.org/about
https://keys.openpgp.org/about/usage#gnupg-upload
https://lists.gnupg.org/pipermail/gnupg-users/2021-August/065384.html

Therefore, the issue with keys.openpgp.org can probably be fixed.

**Comment by danger89** [comment]
It can be fixed. Please, @mpapis @pkuczynski make your email address available if this is indeed the root-cause.

**Comment by dkrugman** [comment]
Yes - fix it by going to the link: https://keys.openpgp.org/upload 

Upload your key file (again if you did it before).

Request the verification email and then click the link in the email "To let others find this key from your email address."
